### ISSUE TITLE
Adaptor does not build under node 16

### ISSUE THREAD

**Issue by Grothesk242** [peterbaumert/ioBroker.vodafone-speedtest#13]
Adaptor does not build under node16. 
Maybe update the deprecated dependencies to more recent versions. 
chokidar also needs to be updated. 

```
pi@raspberrypi:/opt/iobroker/node_modules/iobroker.vodafone-speedtest $ npm install
npm WARN deprecated node-pre-gyp@0.15.0: Please upgrade to @mapbox/node-pre-gyp: the non-scoped node-pre-gyp package is deprecated and only the @mapbox scoped package will recieve updates in the future
npm WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
npm WARN deprecated har-validator@5.1.5: this library is no longer supported
npm WARN deprecated fsevents@1.2.13: fsevents 1 will break on node v14+ and could be using insecure binaries. Upgrade to fsevents 2.
npm WARN deprecated fsevents@2.1.3: "Please update to latest v2.3 or v2.2"

> node-libcurl@2.3.2 install /opt/iobroker/node_modules/iobroker.vodafone-speedtest/node_modules/node-libcurl
> node-pre-gyp install --fallback-to-build

node-pre-gyp WARN Using request for node-pre-gyp https download
node-pre-gyp WARN Tried to download(404): https://github.com/JCMais/node-libcurl/releases/download/v2.3.2/node_libcurl-v2.3.2-node-v93-linux-arm-glibc.tar.gz
node-pre-gyp WARN Pre-built binaries not found for node-libcurl@2.3.2 and node@16.0.0 (node-v93 ABI, glibc) (falling back to source compile with node-gyp)
(node:30076) [DEP0150] DeprecationWarning: Setting process.config is deprecated. In the future the property will be read-only.
(Use `node --trace-deprecation ...` to show where the warning was created)
make: Verzeichnis „/opt/iobroker/node_modules/iobroker.vodafone-speedtest/node_modules/node-libcurl/build“ wird betreten
  CXX(target) Release/obj.target/node_libcurl/src/node_libcurl.o
In file included from /home/iobroker/.cache/node-gyp/16.0.0/include/node/v8.h:30,
                 from /home/iobroker/.cache/node-gyp/16.0.0/include/node/node.h:63,
                 from ../../nan/nan.h:56,
                 from ../src/Curl.h:13,
                 from ../src/node_libcurl.cc:7:
/home/iobroker/.cache/node-gyp/16.0.0/include/node/v8-internal.h: In function ‘void v8::internal::PerformCastCheck(T*)’:
/home/iobroker/.cache/node-gyp/16.0.0/include/node/v8-internal.h:452:38: error: ‘remove_cv_t’ is not a member of ‘std’
             !std::is_same<Data, std::remove_cv_t<T>>::value>::Perform(data);
                                      ^~~~~~~~~~~
/home/iobroker/.cache/node-gyp/16.0.0/include/node/v8-internal.h:452:38: note: suggested alternative: ‘remove_cv’
             !std::is_same<Data, std::remove_cv_t<T>>::value>::Perform(data);
                                      ^~~~~~~~~~~
                                      remove_cv
/home/iobroker/.cache/node-gyp/16.0.0/include/node/v8-internal.h:452:38: error: ‘remove_cv_t’ is not a member of ‘std’
/home/iobroker/.cache/node-gyp/16.0.0/include/node/v8-internal.h:452:38: note: suggested alternative: ‘remove_cv’
             !std::is_same<Data, std::remove_cv_t<T>>::value>::Perform(data);
                                      ^~~~~~~~~~~
                                      remove_cv
/home/iobroker/.cache/node-gyp/16.0.0/include/node/v8-internal.h:452:50: error: template argument 2 is invalid
             !std::is_same<Data, std::remove_cv_t<T>>::value>::Perform(data);
                                                  ^
/home/iobroker/.cache/node-gyp/16.0.0/include/node/v8-internal.h:452:63: error: ‘::Perform’ has not been declared
             !std::is_same<Data, std::remove_cv_t<T>>::value>::Perform(data);
                                                               ^~~~~~~
/home/iobroker/.cache/node-gyp/16.0.0/include/node/v8-internal.h:452:63: note: suggested alternative: ‘herror’
             !std::is_same<Data, std::remove_cv_t<T>>::value>::Perform(data);
                                                               ^~~~~~~
                                                               herror
In file included from ../src/Curl.h:13,
                 from ../src/node_libcurl.cc:7:
../../nan/nan.h: In function ‘void Nan::AsyncQueueWorker(Nan::AsyncWorker*)’:
../../nan/nan.h:2294:62: warning: cast between incompatible function types from ‘void (*)(uv_work_t*)’ {aka ‘void (*)(uv_work_s*)’} to ‘uv_after_work_cb’ {aka ‘void (*)(uv_work_s*, int)’} [-Wcast-function-type]
     , reinterpret_cast<uv_after_work_cb>(AsyncExecuteComplete)
                                                              ^
../src/node_libcurl.cc: In function ‘void NodeLibcurl::Init(Nan::ADDON_REGISTER_FUNCTION_ARGS_TYPE)’:
../src/node_libcurl.cc:61:16: error: cannot convert ‘void (*)(void*)’ to ‘node::Environment*’
   node::AtExit(AtExitCallback, NULL);
                ^~~~~~~~~~~~~~
In file included from ../../nan/nan.h:56,
                 from ../src/Curl.h:13,
                 from ../src/node_libcurl.cc:7:
/home/iobroker/.cache/node-gyp/16.0.0/include/node/node.h:898:38: note:   initializing argument 1 of ‘void node::AtExit(node::Environment*, void (*)(void*), void*)’
 NODE_EXTERN void AtExit(Environment* env,
                         ~~~~~~~~~~~~~^~~
../src/node_libcurl.cc: At global scope:
/home/iobroker/.cache/node-gyp/16.0.0/include/node/node.h:806:43: warning: cast between incompatible function types from ‘void (*)(Nan::ADDON_REGISTER_FUNCTION_ARGS_TYPE)’ {aka ‘void (*)(v8::Local<v8::Object>)’} to ‘node::addon_register_func’ {aka ‘void (*)(v8::Local<v8::Object>, v8::Local<v8::Value>, void*)’} [-Wcast-function-type]
       (node::addon_register_func) (regfunc),                          \
                                           ^
/home/iobroker/.cache/node-gyp/16.0.0/include/node/node.h:840:3: note: in expansion of macro ‘NODE_MODULE_X’
   NODE_MODULE_X(modname, regfunc, NULL, 0)  // NOLINT (readability/null_usage)
   ^~~~~~~~~~~~~
../src/node_libcurl.cc:65:1: note: in expansion of macro ‘NODE_MODULE’
 NODE_MODULE(node_libcurl, Init);
 ^~~~~~~~~~~
make: *** [node_libcurl.target.mk:125: Release/obj.target/node_libcurl/src/node_libcurl.o] Fehler 1
make: Verzeichnis „/opt/iobroker/node_modules/iobroker.vodafone-speedtest/node_modules/node-libcurl/build“ wird verlassen
gyp ERR! build error
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:194:23)
gyp ERR! stack     at ChildProcess.emit (node:events:365:28)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (node:internal/child_process:290:12)
gyp ERR! System Linux 5.10.17-v7l+
gyp ERR! command "/usr/bin/node" "/usr/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "build" "--fallback-to-build" "--module=/opt/iobroker/node_modules/iobroker.vodafone-speedtest/node_modules/node-libcurl/lib/binding/node_libcurl.node" "--module_name=node_libcurl" "--module_path=/opt/iobroker/node_modules/iobroker.vodafone-speedtest/node_modules/node-libcurl/lib/binding" "--napi_version=8" "--node_abi_napi=napi" "--napi_build_version=0" "--node_napi_label=node-v93"
gyp ERR! cwd /opt/iobroker/node_modules/iobroker.vodafone-speedtest/node_modules/node-libcurl
gyp ERR! node -v v16.0.0
gyp ERR! node-gyp -v v5.1.0
gyp ERR! not ok
node-pre-gyp ERR! build error
node-pre-gyp ERR! stack Error: Failed to execute '/usr/bin/node /usr/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js build --fallback-to-build --module=/opt/iobroker/node_modules/iobroker.vodafone-speedtest/node_modules/node-libcurl/lib/binding/node_libcurl.node --module_name=node_libcurl --module_path=/opt/iobroker/node_modules/iobroker.vodafone-speedtest/node_modules/node-libcurl/lib/binding --napi_version=8 --node_abi_napi=napi --napi_build_version=0 --node_napi_label=node-v93' (1)
node-pre-gyp ERR! stack     at ChildProcess.<anonymous> (/opt/iobroker/node_modules/iobroker.vodafone-speedtest/node_modules/node-pre-gyp/lib/util/compile.js:83:29)
node-pre-gyp ERR! stack     at ChildProcess.emit (node:events:365:28)
node-pre-gyp ERR! stack     at maybeClose (node:internal/child_process:1067:16)
node-pre-gyp ERR! stack     at Process.ChildProcess._handle.onexit (node:internal/child_process:301:5)
node-pre-gyp ERR! System Linux 5.10.17-v7l+
node-pre-gyp ERR! command "/usr/bin/node" "/opt/iobroker/node_modules/iobroker.vodafone-speedtest/node_modules/.bin/node-pre-gyp" "install" "--fallback-to-build"
node-pre-gyp ERR! cwd /opt/iobroker/node_modules/iobroker.vodafone-speedtest/node_modules/node-libcurl
node-pre-gyp ERR! node -v v16.0.0
node-pre-gyp ERR! node-pre-gyp -v v0.15.0
node-pre-gyp ERR! not ok
Failed to execute '/usr/bin/node /usr/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js build --fallback-to-build --module=/opt/iobroker/node_modules/iobroker.vodafone-speedtest/node_modules/node-libcurl/lib/binding/node_libcurl.node --module_name=node_libcurl --module_path=/opt/iobroker/node_modules/iobroker.vodafone-speedtest/node_modules/node-libcurl/lib/binding --napi_version=8 --node_abi_napi=napi --napi_build_version=0 --node_napi_label=node-v93' (1)
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@~2.3.1 (node_modules/chokidar/node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@2.3.2: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"arm"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@^1.2.7 (node_modules/glob-watcher/node_modules/chokidar/node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.13: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"arm"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@~2.1.1 (node_modules/mocha/node_modules/chokidar/node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@2.1.3: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"arm"})

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! node-libcurl@2.3.2 install: `node-pre-gyp install --fallback-to-build`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the node-libcurl@2.3.2 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/iobroker/.npm/_logs/2021-04-21T20_00_04_152Z-debug.log
```

**Comment by peterbaumert** [comment]
Yes in future that should be done. Right now iobroker itself only is supported under 10,12 and 14.

**Comment by peterbaumert** [comment]
https://forum.iobroker.net/topic/44421/iobroker-und-nodejs-16

**Comment by Grothesk242** [comment]
No hurry! I'm just raising my hand early so code can be adopted to upcoming releases of nodeJS.

**Comment by peterbaumert** [comment]
@Grothesk242 could you try gain with latest master? The github actions did compile without error now.

**Comment by Grothesk242** [comment]
Yes, seems to run flawless now: 

```
2021-05-17 18:53:00.121  - info: host.chet instance system.adapter.vodafone-speedtest.0 started with pid 16973
2021-05-17 18:53:00.121  - info: host.chet instance system.adapter.vodafone-speedtest.0 started with pid 16973
2021-05-17 18:53:01.845  - info: vodafone-speedtest.0 (16973) starting. Version 0.0.6 in /opt/iobroker/node_modules/iobroker.vodafone-speedtest, node: v16.1.0, js-controller: 3.3.10
2021-05-17 18:53:01.845  - info: vodafone-speedtest.0 (16973) starting. Version 0.0.6 in /opt/iobroker/node_modules/iobroker.vodafone-speedtest, node: v16.1.0, js-controller: 3.3.10
2021-05-17 18:53:03.092  - info: vodafone-speedtest.0 (16973) State value to set for "vodafone-speedtest.0.ip" has to be type "number" but received type "string"
2021-05-17 18:53:03.092  - info: vodafone-speedtest.0 (16973) State value to set for "vodafone-speedtest.0.ip" has to be type "number" but received type "string"
2021-05-17 18:53:30.782  - info: vodafone-speedtest.0 (16973) Vodafone-Speedtest finished with 79.893mbit download speed and 5.585mbit upload speed.
2021-05-17 18:53:30.787  - info: vodafone-speedtest.0 (16973) cleaned everything up...
2021-05-17 18:53:30.841  - info: vodafone-speedtest.0 (16973) Terminated (NO_ERROR): Without reason
2021-05-17 18:53:30.885  - info: vodafone-speedtest.0 (16973) State value to set for "vodafone-speedtest.0.Results.Last_Run" has to be type "number" but received type "object"
2021-05-17 18:53:31.388  - info: host.chet instance system.adapter.vodafone-speedtest.0 terminated with code 0 (NO_ERROR)
2021-05-17 18:53:30.782  - info: vodafone-speedtest.0 (16973) Vodafone-Speedtest finished with 79.893mbit download speed and 5.585mbit upload speed.
2021-05-17 18:53:30.787  - info: vodafone-speedtest.0 (16973) cleaned everything up...
2021-05-17 18:53:30.841  - info: vodafone-speedtest.0 (16973) Terminated (NO_ERROR): Without reason
2021-05-17 18:53:30.885  - info: vodafone-speedtest.0 (16973) State value to set for "vodafone-speedtest.0.Results.Last_Run" has to be type "number" but received type "object"
2021-05-17 18:53:31.388  - info: host.chet instance system.adapter.vodafone-speedtest.0 terminated with code 0 (NO_ERROR)
```

Histroy adapter gets feed with values and eCharts is drawing again. 
Only 
```info: vodafone-speedtest.0 (16973) State value to set for "vodafone-speedtest.0.Results.Last_Run" has to be type "number" but received type "object"```
could be adressed, maybe?

My Test environment: 
```
node v16.1.0
npm 7.13.0
js-controller 3.3.10
iobroker.vodafone-speedtest@0.0.6 (git+ssh://git@github.com/peterbaumert/ioBroker.vodafone-speedtest.git#b7aa0149435ee6591b6b9ddee202591f200e4e20)
```

**Comment by Grothesk242** [comment]
Works with node v14.17.0

**Comment by Grothesk242** [comment]
vodafone-speedtest@0.0.6 (git+ssh://git@github.com/peterbaumert/ioBroker.vodafone-speedtest.git#51a7b2cd9a24f14e6935d0dc805d02a71d8aaf87)

I now silent in regard of the types 
`value XYZ has to be type "this" but received type "that"`
does not show in the log

**Comment by peterbaumert** [comment]
fixed in 0.7.0
